packages/fxa-auth-server/test/scripts/prune-tokens.js

```diff
@@ -141,20 +159,134 @@ describe('scripts/prune-tokens', () => {
       {
         cwd,
         shell: '/bin/bash',
       }
     );
-    assert.isTrue(/"@passwordForgotTokensDeleted":1/.test(stderr));
-    assert.isTrue(/"@passwordChangeTokensDeleted":1/.test(stderr));
-    assert.isTrue(/"@accountResetTokensDeleted":1/.test(stderr));
-    assert.isTrue(/"@sessionTokensDeleted":1/.test(stderr));
-    assert.isTrue(/"@unblockCodesDeleted":1/.test(stderr));
-    assert.isTrue(/"@signInCodesDeleted":1/.test(stderr));
+    assert.isTrue(/skipping limit sessions operation./.test(stderr));
+    assert.isTrue(/skipping token pruning operation./.test(stderr));
+  });
+
+  describe('prune tokens', () => {
+    before(async () => {
+      await clearDb();
+
+      await SessionToken.create(sessionToken());
+      await PasswordChangeToken.create(passwordChangeToken);
+      await PasswordForgotToken.create(passwordForgotToken);
+      await AccountResetToken.knexQuery().insert(accountResetToken);
+      await UnblockCodes.knexQuery().insert(unblockCode);
+      await SignInCodes.knexQuery().insert(signInCode);
+    });
+
+    after(async () => {
+      await clearDb();
+    });
+
+    it('prunes tokens', async () => {
+      // Note that logger output, directs to standard err.
+      const { stderr } = await exec(
+        `NODE_ENV=dev node -r esbuild-register scripts/prune-tokens.ts '--maxTokenAge=${maxAge}-days' '--maxCodeAge=${maxAge}-days' `,
+        {
+          cwd,
+          shell: '/bin/bash',
+        }
+      );
+      assert.isTrue(/"@passwordForgotTokensDeleted":1/.test(stderr));
+      assert.isTrue(/"@passwordChangeTokensDeleted":1/.test(stderr));
+      assert.isTrue(/"@accountResetTokensDeleted":1/.test(stderr));
+      assert.isTrue(/"@sessionTokensDeleted":1/.test(stderr));
+      assert.isTrue(/"@unblockCodesDeleted":1/.test(stderr));
+      assert.isTrue(/"@signInCodesDeleted":1/.test(stderr));
+    });
+  });
+
+  describe('limits sessions', async () => {
+    const size = 10;
+    let tokens = [];
+    let devices = [];
+
+    const sessionAt = (i) => SessionToken.findByTokenId(tokens.at(i).id);
+
+    const deviceAt = (i) =>
+      Device.findByPrimaryKey(devices.at(i).uid, devices.at(i).id);
+
+    const sessionCount = async () =>
+      (await SessionToken.knexQuery().count())[0]['count(*)'];
+
+    const deviceCount = async () =>
+      (await Device.knexQuery().count())[0]['count(*)'];
+
+    before(async () => {
+      await clearDb();
+      tokens = [];
+      devices = [];
+
+      // Make sure db state is clean
+      await SessionToken.knexQuery().del();
+      await Device.knexQuery().del();
+
+      // Add tokens
+      for (let i = 0; i < size; i++) {
+        const curToken = sessionToken();
+        const curDevice = device(account.uid, curToken.id);
+        curToken.createdAt = Date.now();
+
+        await SessionToken.create(curToken);
+        await Device.create(curDevice);
+        await new Promise((r) => setTimeout(r, 10));
```

Comment:
What is this for?

Reply:
Ensures that the createdAt timestamps are spaced out a bit.